starlette/applications.py

```diff
@@ -138,7 +138,7 @@ def host(
         self.router.host(host, app=app, name=name)
 
     def add_middleware(
-        self, middleware_class: type, **options: typing.Any
+        self, middleware_class: typing.Type[MiddlewareType], **options: typing.Any
```

Comment:
Rather than introducing a new type alias, is there a reason not to reuse `ASGIApp` here? As in, `middleware_class: Type[ASGIApp]`

A middleware instance _is_ an ASGI application, so that would make sense to me.

Reply:
Please do. :pray: